src/WebJobs.Script.WebHost/WebScriptHostManager.cs

```diff
@@ -418,7 +418,7 @@ internal static void AddRouteDataToRequest(IHttpRouteData routeData, HttpRequest
                     routeDataValues.Add(pair.Key, value);
                 }
 
-                request.Properties.Add(HttpExtensionConstants.AzureWebJobsHttpRouteDataKey, routeDataValues);
+                request.Properties[HttpExtensionConstants.AzureWebJobsHttpRouteDataKey] = routeDataValues;
```

Comment:
Does this mean we're setting property multiple times now?

Reply:
Yes. If the call goes from proxies to functions. the request uri changes (as the backend changes) and this means that the routedata will need to be updated.

---
In reply to: [136209751](https://github.com/Azure/azure-webjobs-sdk-script/pull/1839#discussion_r136209751) [](ancestors = 136209751)

Reply:
You should add a new test for this since you rely on this new behavior and don't want it to regress. See the unit tests we already have in place for this method.

Reply:
Good point, I will update the existing test to make sure this property is always set when routeData.Values != null